crates/swc_ecma_minifier/src/analyzer/mod.rs

```diff
@@ -1070,15 +1085,23 @@ where
     }
 
     #[cfg_attr(feature = "debug", tracing::instrument(skip(self, n)))]
-    fn visit_switch_case(&mut self, n: &SwitchCase) {
-        n.test.visit_with(self);
+    fn visit_switch_stmt(&mut self, n: &SwitchStmt) {
+        n.discriminant.visit_with(self);
 
-        {
+        let mut fallthrough = false;
+
+        for case in n.cases.iter() {
             let ctx = Ctx {
                 in_cond: true,
                 ..self.ctx
             };
-            self.with_ctx(ctx).visit_in_cond(&n.cons);
+            if fallthrough {
```

Comment:
Is this `if` required?

Reply:
Yes. If fallthrough, case body won't be able to see case test, in other cases they could
```js
case 1:
case a = 1:
   console.log(a)
```
vs
```js
case 1:
   break
case a = 1:
   console.log(a)
```

Reply:
Thanks!
I looked at the def of `visit_in_cond` and found that this is required.